packages/playground/blueprints/src/lib/steps/install-plugin.ts

```diff
@@ -58,110 +61,134 @@ export const installPlugin: StepHandler<InstallPluginStep<File>> = async (
-			url: '/wp-admin/update.php?action=upload-plugin',
-			method: 'POST',
-			formData: postData,
-			files: { pluginzip: pluginZipFile },
+		await unzip(playground, {
+			zipPath: tmpZipPath,
+			extractToPath: tmpFolder,
 		});
 
-		// Activate if needed
-		if (activate) {
-			const pluginInstalledPage = asDOM(pluginInstalledResponse);
-			const activateButtonHref = pluginInstalledPage
-				.querySelector('#wpbody-content .button.button-primary')!
-				.attributes.getNamedItem('href')!.value;
-			const activatePluginUrl = new URL(
-				activateButtonHref,
-				await playground.pathToInternalUrl('/wp-admin/')
-			).toString();
-			await playground.request({
-				url: activatePluginUrl,
-			});
+		await playground.unlink(tmpZipPath);
+
+		// Find extracted plugin folder name
+
+		const files = await playground.listFiles(tmpFolder);
+
+		let pluginFolderName;
+		let tmpPluginPath = '';
+
+		for (const file of files) {
+			tmpPluginPath = `${tmpFolder}/${file}`;
+			if (await playground.isDir(tmpPluginPath)) {
+				pluginFolderName = file;
+				break;
+			}
 		}
 
-		/**
-		 * Pair the site editor's nested iframe to the Service Worker.
-		 *
-		 * Without the patch below, the site editor initiates network requests that
-		 * aren't routed through the service worker. That's a known browser issue:
-		 *
-		 * * https://bugs.chromium.org/p/chromium/issues/detail?id=880768
-		 * * https://bugzilla.mozilla.org/show_bug.cgi?id=1293277
-		 * * https://github.com/w3c/ServiceWorker/issues/765
-		 *
-		 * The problem with iframes using srcDoc and src="about:blank" as they
-		 * fail to inherit the root site's service worker.
-		 *
-		 * Gutenberg loads the site editor using <iframe srcDoc="<!doctype html">
-		 * to force the standards mode and not the quirks mode:
-		 *
-		 * https://github.com/WordPress/gutenberg/pull/38855
-		 *
-		 * This commit patches the site editor to achieve the same result via
-		 * <iframe src="/doctype.html"> and a doctype.html file containing just
-		 * `<!doctype html>`. This allows the iframe to inherit the service worker
-		 * and correctly load all the css, js, fonts, images, and other assets.
-		 *
-		 * Ideally this issue would be fixed directly in Gutenberg and the patch
-		 * below would be removed.
-		 *
-		 * See https://github.com/WordPress/wordpress-playground/issues/42 for more details
-		 */
-		if (
-			(await playground.isDir(
-				'/wordpress/wp-content/plugins/gutenberg'
-			)) &&
-			!(await playground.fileExists('/wordpress/.gutenberg-patched'))
-		) {
-			await playground.writeFile('/wordpress/.gutenberg-patched', '1');
-			await updateFile(
-				playground,
-				`/wordpress/wp-content/plugins/gutenberg/build/block-editor/index.js`,
-				(contents) =>
-					contents.replace(
-						/srcDoc:("[^"]+"|[^,]+)/g,
-						'src:"/wp-includes/empty.html"'
-					)
-			);
-			await updateFile(
+		if (!pluginFolderName) {
+			throw new Error('Extracted plugin folder not found');
```

Comment:
Let’s be a bit more verbose with the error messages:

```suggestion
			throw new Error(`The plugin zip file should contain a folder with plugin files inside, but the provided zip file (${zipFileName}) does not contain such a folders.`);
```